src/util/usePaginatedQuery.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useEffect, useState } from "react";
+import useDebounce from "./useDebounce";
+
+interface UseDelayedQuery {
+  defaultQuery?: string;
+  defaultPage?: number;
+  debounceTime?: number;
+}
+
+export const usePaginatedQuery = ({ defaultQuery = "", defaultPage = 1, debounceTime = 300 }: UseDelayedQuery = {}) => {
+  const [query, _setQuery] = useState(defaultQuery ?? "");
+  const [page, setPage] = useState(defaultPage);
+  const delayedQuery = useDebounce(query, debounceTime ?? 300);
```

Comment:
Både debounceTime og defaultQuery har jo default verdier, så kan fjerne fallback-verdi herfra og fra initiell state for query